worldedit-core/src/main/java/com/sk89q/worldedit/regions/Region.java

```diff
@@ -61,8 +61,22 @@
      * Get the number of blocks in the region.
      *
      * @return number of blocks
+     * @deprecated use {@link Region#getVolume()} to prevent overflows
      */
-    int getArea();
+    @Deprecated
+    default int getArea() {
+        return (int) getVolume();
```

Comment:
This is a highly dangerous setup -- I suggest we add some sort of check on `getVolume()` to indicate the problem instead of letting it crash with a `StackOverflowError` if someone overrides neither `getArea()` nor `getVolume()`.

We can reflect onto the current class and check if the `clazz.getMethod("getArea")` points to this method, defined in `Region` -- this will catch it being overriden at any point in the hierarchy.

Having this check in `getVolume()` shouldn't be a huge deal, because for all the core classes, we'll have overriden `getVolume()` and the check will never happen. It's just a compatibility for older mods that might be implementing `Region`.